### PR TITLE
Make the resource fetching of all services concurrent

### PR DESCRIPTION
**Pull request by misg**
This PR brings the final enhancement about concurrency in our refactoring (#183) by fetching all the services of a cloud provider concurrently.
I tested it with an execution against AWS (all services, privates included) and it took 1min40 to build the report, instead of 2min30 for the version of Scout Suite on develop.
Definitely encouraging!

### CODE REVIEW

File: ScoutSuite/providers/base/configs/services.py [hunk 1]
```diff
@@ -16,28 +16,35 @@ def _is_provider(self, provider_name):
...
+        tasks = {
+            asyncio.ensure_future(
+                self._fetch(service, services, regions)
+            ) for service in vars(self)
+        }
+        await asyncio.wait(tasks)
+
+    async def _fetch(self, service, services, regions):
+        try:
+            # skip services
```
Comment (Aboisier) [inline ScoutSuite/providers/base/configs/services.py]: Est-ce qu'on pourrait skip les services dans `fetch` plutôt qu'ici? Ça éviterait d'avoir à passer `services` pour rien.

**Comment by j4v** [comment]
> it took 1min40 to build the report, instead of 2min30 for the version of Scout Suite on develop

Is this using the same amount of threads/workers?

**Comment by misg** [comment]
> > it took 1min40 to build the report, instead of 2min30 for the version of Scout Suite on develop
> 
> Is this using the same amount of threads/workers?

It was using 10 threads on our refactorized version and I thought it was the same for the version on `develop` but according to https://github.com/nccgroup/ScoutSuite/blob/develop/ScoutSuite/providers/aws/configs/regions.py#L134 it seems like it was actually using 20 threads (10 for fetching and 10 for parsing, by default (I didn't pass a `thread-config` argument))...

File: ScoutSuite/providers/base/configs/services.py (hunk 1 above)
Comment (misg) [reply ScoutSuite/providers/base/configs/services.py]: Done!

**Pull request merged**